hail/src/main/scala/is/hail/shuffler/ShuffleServer.scala

```diff
@@ -0,0 +1,234 @@
+package is.hail.shuffler
+
+import is.hail.shuffler.ShuffleUtils._
+import org.apache.log4j.Logger;
+import is.hail.HailLSM
+import is.hail.annotations.{ Region, UnsafeRow }
+import is.hail.expr.ir.IRParser
+import is.hail.expr.types.encoded.EType
+import is.hail.expr.types.physical.{ PStruct, PType }
+import is.hail.expr.types.virtual.{ TStruct, Type }
+import is.hail.io.TypedCodecSpec
+import is.hail.rvd.AbstractRVDSpec
+import java.io._
+import java.net._
+import java.security.KeyStore;
+import java.util.UUID
+import java.util.concurrent.{ ConcurrentSkipListMap, Executors }
+import javax.net._
+import javax.net.ssl._
+import javax.security.cert.X509Certificate;
+import java.util.concurrent._
+import org.json4s.jackson.{ JsonMethods, Serialization }
+import java.nio.charset.StandardCharsets
+
+import scala.annotation.switch
+
+import is.hail.utils._
+
+class Handler (
+  val server: ShuffleServer,
+  val socket: Socket
+) extends Runnable {
+  val log = Logger.getLogger(this.getClass.getName());
+  val in = new DataInputStream(socket.getInputStream)
+  val out = new DataOutputStream(socket.getOutputStream)
+
+  def run(): Unit = {
+    try {
+      log.info(s"SERV handle")
+      try {
+        var continue = true
+        while (continue) {
+          val op = in.read()
+          log.info(s"SERV operation ${op}")
+            (op: @switch) match {
+            case Wire.START => start()
+            case Wire.PUT => put()
+            case Wire.GET => get()
+            case -1 =>
+              log.info(s"client closed socket, exiting cleanly")
+              continue = false
+            case op => fatal(s"bad operation number $op")
+          }
+        }
+      } finally {
+        socket.close()
+      }
+    } catch {
+      case e: Exception =>
+        println(s"exception while serving ${e.getMessage}")
+        log.info(s"exception while serving", e)
+    }
+  }
+
+  def readShuffleUUID(): Shuffle = {
+    val uuid = in.readUTF()
+    val shuffle = server.shuffles.get(uuid)
+    if (shuffle == null) {
+      throw new RuntimeException(s"shuffle does not exist $uuid")
```

Comment:
This doesn't seem right.  It looks like this will throw into the handler which will close the socket connect.  The wire protocol should have a response including errors.